rollup.config.js

```diff
@@ -10,6 +10,22 @@ const terser = require('rollup-plugin-terser').terser;
 const pkg = require('./package.json');
 
 const input = 'src/index.js';
+const inputESM = {
+	'dist/chart.esm': 'src/index.esm.js',
+	'helpers/canvas': 'src/helpers/helpers.canvas.js',
```

Comment:
Maybe we could programatically build the rollup input instead of having to list every file?

Though this is pretty explicit, so it would still be helpful to have a comment showing exactly what it's doing if done programatically

Reply:
Is this even necessary to do programatically on the long term?

If `esm_buid.zip` is not anticipating the future repo structure, I do not see the benefit to generate an additional multi-file ESM package, which basically restructures `src/`, instead of delivering `src/` directly, as it already is a fully working ESM package.

Reply:
The problem is, it is not a fully working esm module. Because there are proposal features used.

Reply:
Oh I see, wasn't aware of this.

If it is in order to pull the source through babel, we could also preserve the current structure and wouldn't need to define a restructured package.

However, I am unsure if all this effort is worth it to build ChartJS on top of non-JS syntax/features. I assume that you already need to babel the source to apply ESLint (which only supports Stage 4) and your testing environment?

Is the only used proposal feature the *out of constructor*-class property initialisation from `@babel/plugin-proposal-class-properties`? `Object.assign` and `static` methods  should be Stage 4 since 2015? If we only need to move the property initialisation into the class constructors, in order to avoid all this additional tooling, building (and the whole maintenance behind it), I believe it would be worth it 😃

Reply:
Yeah, it's just the `static` initializations. It wouldn't be hard to change. I wasn't fully aware of the implications when I originally introduced it and would definitely be in favor of removing it and babel from the ES builds

Reply:
I'm on board with removing the static initializations to remove the need for babel in the ESM build. We should add some documentation to `docs/developers/contributing.md` with these guidelines otherwise I'm sure in a year we'll forget and add back something that's not stage 4.

Reply:
We still need the tooling for IE support, so it does not really reduce any maintenance. But I created a request for moving the statics to module scope: #7425

Reply:
Wouldn't it remove the need to babel everything during testing and linting (as well as the dependencies necessary for this)?

Since IE does not support the ESM syntax, I assume that at most IE-compatible packages need Babel after their generation?

I would also suggest to never apply Babel to the ESM package. This would allow users to downgrade the JS standard of ChartJS (which can have quite the performance impact) based on their own needs.

Reply:
> Maybe we could programatically build the rollup input instead of having to list every file?

Added glob for that.

Reply:
@sebiniemann Chart.js v3 `umd` bundle is IE compatible and we need babel for that. Its produced from the same source. Some tests use ESM syntax, so we need to rollup those so karma can cope with them. We also need babel for the tests to get coverage right (I've tried to get it done with rollup-plugin-istanbul and several other packages, but it just does not work right).

Reply:
I am 100% with you on this :)

Not sure why babel is needed for test coverage, might be some special configuration. We could also remove `babel-eslint` (hard to see, as it comes as a sub-dep of `eslint-config-esnext`).

Babel will surly remain in most projects to deliver at least one *fast-to-use* version with guaranteed browser compatibility (which is nice), but as testing and linting is not bound to use land, it shouldn't be necessary for these.